File: python/VariableLengthQuantity.py
vlq_num_mask = 0b1111111
vlq_continue_mask = 1<<7
def encode(numbers):
    # 1-liner 
    return [((number>>(7*i)) & vlq_num_mask) | (bool(i)<<7)
            for number in numbers
            for i in range(((number>>1).bit_length())//7, -1, -1)]
def decode(bytes_):
    return list(_decode(bytes_))
def _decode(bytes_):
    code = 0
    if bytes_[-1] & vlq_continue_mask:
        raise ValueError('incomplete sequence')
    for byte in bytes_:
        code = (code << 7)|(byte & vlq_num_mask)
        if not(byte & vlq_continue_mask):
            yield code
            code = 0

File: javascript/VariableLengthQuantity.js
export const encode = (uint32s) => {
  return uint32s.map((elm) => {
    if (elm === 0) return [0];
    let bytes = [];
    while (elm > 0) {
      let byte = 0x7f & elm;
      if (bytes.length != 0) byte |= 0x80;
      bytes = [byte, ...bytes];
      elm >>>= 7;
    }
    return bytes;
  }).reduce((out, elm) => [...out, ...elm]);
};
export const decode = (bytes) => {
  if ((0x80 & bytes[bytes.length - 1]) != 0) throw new Error('Incomplete sequence');
  let i = 0;
  return bytes.reduce((out, byte) => {
    i <<= 7;
    i |= 0x7f & byte;
    if ((0x80 & byte) == 0) {
      const res = i >>> 0;
      i = 0;
      return [...out, res];
    }
    return out;
  },[]);
};